Tools/TestWebKitAPI/cocoa/TestWKWebView.mm

```diff
@@ -264,26 +279,40 @@ - (void)extendSelectionToEndOfParagraph
 {
     TestWebKitAPI::AutocorrectionContext result;
     bool done = false;
-    [self.textInputContentView requestAutocorrectionContextWithCompletionHandler:[&] (UIWKAutocorrectionContext *context) {
-        result = { context.contextBeforeSelection, context.selectedText, context.contextAfterSelection };
-        done = true;
-    }];
+#if HAVE(UI_ASYNC_TEXT_INTERACTION)
+    if (auto asyncTextInput = self.asyncTextInput) {
+        [asyncTextInput requestTextContextForAutocorrectionWithCompletionHandler:[&](WKSETextDocumentContext *context) {
+            result = {
+                dynamic_objc_cast<NSString>(context.contextBefore),
+                dynamic_objc_cast<NSString>(context.selectedText),
+                dynamic_objc_cast<NSString>(context.contextAfter),
+            };
+            done = true;
+        }];
+    } else
+#endif // HAVE(UI_ASYNC_TEXT_INTERACTION)
+    {
+        [self.textInputContentView requestAutocorrectionContextWithCompletionHandler:[&](UIWKAutocorrectionContext *context) {
+            result = { context.contextBeforeSelection, context.selectedText, context.contextAfterSelection };
+            done = true;
+        }];
+    }
     TestWebKitAPI::Util::run(&done);
     return result;
 }
 
 - (id<UITextInputTraits_Private>)effectiveTextInputTraits
 {
-#if HAVE(UI_ASYNC_TEXT_INPUT)
+#if HAVE(UI_ASYNC_TEXT_INTERACTION)
     if (self.hasAsyncTextInput)
-        return static_cast<id<UITextInputTraits_Private>(self.extendedTextInputTraits);
+        return static_cast<id<UITextInputTraits_Private>>(self.extendedTextInputTraits);
```

Comment:
How did this even compile before??

Reply:
`UI_ASYNC_TEXT_INPUT` should've been `UI_ASYNC_TEXT_INTERACTION`, so this was essentially dead code.

(I audited for other similar mistakes, and found a couple in `WKContentViewInteraction.mm` which are fixed by this patch)